pkg/engine2/operational_eval/evaluator.go

```diff
@@ -452,12 +453,17 @@ func (eval *Evaluator) UpdateId(oldId, newId construct.ResourceId) error {
 				vertex.Edge = UpdateEdgeId(vertex.Edge, oldId, newId)
 				replaceVertex(key, vertex)
 				// because the temp graph contains the src and target as nodes, we need to update it if it exists
-				if vertex.TempGraph != nil {
-					err := construct.ReplaceResource(vertex.TempGraph, oldId, &construct.Resource{ID: newId})
-					if err != nil {
-						errs = errors.Join(errs, err)
-						continue
-					}
+			}
+			if vertex.TempGraph != nil {
+				_, err := vertex.TempGraph.Vertex(oldId)
+				switch {
+				case errors.Is(err, graph.ErrVertexNotFound):
+					// do nothing
+					break
```

Comment:
`break`? that'd break out of the `for _, key := range topo` loop, is that what you intended?

Reply:
nope definitely didnt mean for that to be there, removing